misc/ballerinalang-data-mapper/src/main/java/org/ballerinalang/datamapper/AIDataMapperCodeActionUtil.java

```diff
@@ -106,6 +106,10 @@
         if (props.size() != 2 || props.get(RIGHT_SYMBOL_INDEX).kind() != DiagnosticPropertyKind.SYMBOLIC ||
                 props.get(LEFT_SYMBOL_INDEX).kind() != DiagnosticPropertyKind.SYMBOLIC) {
             return fEdits;
+        } else if ((((TypeSymbol) props.get(0).value()).typeKind() != TypeDescKind.TYPE_REFERENCE) &&
```

Comment:
`if` and `else if` conditions can be merged together

Reply:
Merged the if and else if